File: hcbe-backend/Models/CommunityPrograms.cs
namespace HcbeApi.Models;

public static class CommunityProgramStatuses
{
    public const string Draft = "Draft";
    public const string Submitted = "Submitted";
    public const string Approved = "Approved";
    public const string Rejected = "Rejected";
    public const string Active = "Active";
    public const string Cancelled = "Cancelled";
    public const string Withdrawn = "Withdrawn";
    public const string Published = "Published";
}

public sealed class CommunityBusiness
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid OwnerUserId { get; set; }
    public User? OwnerUser { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? NameEn { get; set; }
    public string Category { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? DescriptionEn { get; set; }
    public string? Services { get; set; }
    public string? ServicesEn { get; set; }
    public string ContactEmail { get; set; } = string.Empty;
    public string? ContactPhone { get; set; }
    public string? WebsiteUrl { get; set; }
    public string? LogoUrl { get; set; }
    public string? City { get; set; }
    public string? Province { get; set; }
    public string? ServiceRegions { get; set; }
    public string Status { get; set; } = CommunityProgramStatuses.Submitted;
    public bool IsFeatured { get; set; }
    public string? ReviewNotes { get; set; }
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? ReviewedAtUtc { get; set; }
}

public sealed class NewcomerJourney
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid UserId { get; set; }
    public User? User { get; set; }
    public DateOnly? ArrivalDate { get; set; }
    public string? City { get; set; }
    public string? Province { get; set; }
    public string PreferredLanguage { get; set; } = "fr";
    public string NeedsJson { get; set; } = "[]";
    public string CompletedStepsJson { get; set; } = "[]";
    public bool MentorRequested { get; set; }
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;
}

public sealed class FamilyHousehold
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid OwnerUserId { get; set; }
    public User? OwnerUser { get; set; }
    public string HouseholdName { get; set; } = string.Empty;
    public string Status { get; set; } = CommunityProgramStatuses.Active;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;
    public ICollection<FamilyHouseholdMember> Members { get; set; } = new List<FamilyHouseholdMember>();
}

public sealed class FamilyHouseholdMember
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid HouseholdId { get; set; }
    public FamilyHousehold? Household { get; set; }
    public string FullName { get; set; } = string.Empty;
    public string Relationship { get; set; } = string.Empty;
    public string? Email { get; set; }
    public DateOnly? BirthDate { get; set; }
    public string Status { get; set; } = CommunityProgramStatuses.Active;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
}

public sealed class AppointmentOffering
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Title { get; set; } = string.Empty;
    public string? TitleEn { get; set; }
    public string Description { get; set; } = string.Empty;
    public string? DescriptionEn { get; set; }
    public string Category { get; set; } = string.Empty;
    public string Mode { get; set; } = "Online";
    public string? Location { get; set; }
    public string? LocationEn { get; set; }
    public int DurationMinutes { get; set; } = 30;
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;
    public ICollection<AppointmentSlot> Slots { get; set; } = new List<AppointmentSlot>();
}

public sealed class AppointmentSlot
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid OfferingId { get; set; }
    public AppointmentOffering? Offering { get; set; }
    public DateTime StartsAtUtc { get; set; }
    public DateTime EndsAtUtc { get; set; }
    public int Capacity { get; set; } = 1;
    public bool IsCancelled { get; set; }
    public ICollection<AppointmentBooking> Bookings { get; set; } = new List<AppointmentBooking>();
}

public sealed class AppointmentBooking
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid SlotId { get; set; }
    public AppointmentSlot? Slot { get; set; }
    public Guid UserId { get; set; }
    public User? User { get; set; }
    public string? Reason { get; set; }
    public string Status { get; set; } = CommunityProgramStatuses.Active;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? CancelledAtUtc { get; set; }
}

public sealed class PartnerBenefit
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid PartnerId { get; set; }
    public Partner? Partner { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? TitleEn { get; set; }
    public string Description { get; set; } = string.Empty;
    public string? DescriptionEn { get; set; }
    public string? Terms { get; set; }
    public string? TermsEn { get; set; }
    public string? RedemptionInstructions { get; set; }
    public string? RedemptionInstructionsEn { get; set; }
    public string? SharedCode { get; set; }
    public DateTime? StartsAtUtc { get; set; }
    public DateTime? EndsAtUtc { get; set; }
    public int? MaxClaims { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;
    public ICollection<PartnerBenefitClaim> Claims { get; set; } = new List<PartnerBenefitClaim>();
}

public sealed class PartnerBenefitClaim
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid BenefitId { get; set; }
    public PartnerBenefit? Benefit { get; set; }
    public Guid UserId { get; set; }
    public User? User { get; set; }
    public string RedemptionCode { get; set; } = string.Empty;
    public string Status { get; set; } = CommunityProgramStatuses.Active;
    public DateTime ClaimedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? RedeemedAtUtc { get; set; }
}

public sealed class GrantApplication
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid GrantProgramId { get; set; }
    public GrantProgram? GrantProgram { get; set; }
    public Guid UserId { get; set; }
    public User? User { get; set; }
    public string ApplicantName { get; set; } = string.Empty;
    public string ApplicantEmail { get; set; } = string.Empty;
    public string Statement { get; set; } = string.Empty;
    public string AnswersJson { get; set; } = "{}";
    public string DocumentsJson { get; set; } = "[]";
    public string Status { get; set; } = CommunityProgramStatuses.Submitted;
    public string? AdminNotes { get; set; }
    public DateTime SubmittedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? ReviewedAtUtc { get; set; }
}

public sealed class SponsorshipPackage
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Title { get; set; } = string.Empty;
    public string? TitleEn { get; set; }
    public string Description { get; set; } = string.Empty;
    public string? DescriptionEn { get; set; }
    public string DeliverablesJson { get; set; } = "[]";
    public long AmountCents { get; set; }
    public string Currency { get; set; } = "cad";
    public bool IsActive { get; set; } = true;
    public int DisplayOrder { get; set; }
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;
}

public sealed class SponsorshipRequest
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid UserId { get; set; }
    public User? User { get; set; }
    public Guid? PackageId { get; set; }
    public SponsorshipPackage? Package { get; set; }
    public string OrganizationName { get; set; } = string.Empty;
    public string ContactEmail { get; set; } = string.Empty;
    public string Objective { get; set; } = string.Empty;
    public string? Notes { get; set; }
    public long ProposedAmountCents { get; set; }
    public string Currency { get; set; } = "cad";
    public string Status { get; set; } = CommunityProgramStatuses.Submitted;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? ReviewedAtUtc { get; set; }
}

public sealed class AnnualCommunityReport
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public int Year { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? TitleEn { get; set; }
    public string Summary { get; set; } = string.Empty;
    public string? SummaryEn { get; set; }
    public string MetricsJson { get; set; } = "{}";
    public string Status { get; set; } = CommunityProgramStatuses.Draft;
    public DateTime GeneratedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? PublishedAtUtc { get; set; }
}

public sealed class OperationalAutomationRule
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Key { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string NameEn { get; set; } = string.Empty;
    public string Cadence { get; set; } = "Daily";
    public bool IsEnabled { get; set; } = true;
    public DateTime? LastRunAtUtc { get; set; }
    public DateTime? NextRunAtUtc { get; set; }
    public string? LastStatus { get; set; }
    public string? LastSummary { get; set; }
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;
}

File: hcbe-frontend/src/lib/api/community-programs.ts
import { apiClient } from './client';

export interface CommunityBusiness { id: string; name: string; nameEn?: string; category: string; description: string; descriptionEn?: string; services?: string; servicesEn?: string; contactEmail: string; contactPhone?: string; websiteUrl?: string; logoUrl?: string; city?: string; province?: string; serviceRegions?: string; status: string; isFeatured: boolean; reviewNotes?: string; createdAtUtc: string; updatedAtUtc: string; }
export interface NewcomerJourney { id: string; arrivalDate?: string; city?: string; province?: string; preferredLanguage: string; needs: string[]; completedSteps: string[]; mentorRequested: boolean; progressPercent: number; updatedAtUtc: string; }
export interface FamilyMember { id: string; fullName: string; relationship: string; email?: string; birthDate?: string; status: string; createdAtUtc: string; }
export interface FamilyHousehold { id: string; householdName: string; status: string; members: FamilyMember[]; updatedAtUtc: string; }
export interface AppointmentOffering { id: string; title: string; titleEn?: string; description: string; descriptionEn?: string; category: string; mode: string; location?: string; locationEn?: string; durationMinutes: number; isActive: boolean; }
export interface AppointmentSlot { id: string; offeringId: string; offeringTitle: string; offeringTitleEn?: string; startsAtUtc: string; endsAtUtc: string; capacity: number; available: number; isCancelled: boolean; }
export interface AppointmentBooking { id: string; slotId: string; offeringTitle: string; offeringTitleEn?: string; startsAtUtc: string; endsAtUtc: string; reason?: string; status: string; createdAtUtc: string; }
export interface PartnerBenefit { id: string; partnerId: string; partnerName: string; partnerLogoUrl?: string; title: string; titleEn?: string; description: string; descriptionEn?: string; terms?: string; termsEn?: string; redemptionInstructions?: string; redemptionInstructionsEn?: string; startsAtUtc?: string; endsAtUtc?: string; maxClaims?: number; claimCount: number; isActive: boolean; isClaimed: boolean; redemptionCode?: string; }
export interface GrantApplication { id: string; grantProgramId: string; programTitle: string; programTitleEn?: string; applicantName: string; applicantEmail: string; statement: string; answers: Record<string, string>; documents: string[]; status: string; adminNotes?: string; submittedAtUtc: string; updatedAtUtc: string; }
export interface SponsorshipPackage { id: string; title: string; titleEn?: string; description: string; descriptionEn?: string; deliverables: string[]; amountCents: number; currency: string; isActive: boolean; displayOrder: number; }
export interface SponsorshipRequest { id: string; packageId?: string; packageTitle?: string; organizationName: string; contactEmail: string; objective: string; notes?: string; proposedAmountCents: number; currency: string; status: string; createdAtUtc: string; updatedAtUtc: string; }
export interface AnnualReport { id: string; year: number; title: string; titleEn?: string; summary: string; summaryEn?: string; metrics: Record<string, number>; status: string; generatedAtUtc: string; publishedAtUtc?: string; }
export interface AutomationRule { id: string; key: string; name: string; nameEn: string; cadence: string; isEnabled: boolean; lastRunAtUtc?: string; nextRunAtUtc?: string; lastStatus?: string; lastSummary?: string; }
export interface ProgramsAdminOverview { businesses: CommunityBusiness[]; offerings: AppointmentOffering[]; slots: AppointmentSlot[]; bookings: AppointmentBooking[]; benefits: PartnerBenefit[]; grantApplications: GrantApplication[]; sponsorshipPackages: SponsorshipPackage[]; sponsorshipRequests: SponsorshipRequest[]; annualReports: AnnualReport[]; automationRules: AutomationRule[]; }

export interface BusinessInput { name: string; nameEn?: string; category: string; description: string; descriptionEn?: string; services?: string; servicesEn?: string; contactEmail: string; contactPhone?: string; websiteUrl?: string; logoUrl?: string; city?: string; province?: string; serviceRegions?: string; }
export interface JourneyInput { arrivalDate?: string; city?: string; province?: string; preferredLanguage: string; needs: string[]; completedSteps: string[]; mentorRequested: boolean; }

export const communityProgramsApi = {
  directory: (search = '', category = '', province = '') => apiClient.get<CommunityBusiness[]>(`/api/community-programs/directory?${new URLSearchParams({ search, category, province })}`),
  reports: () => apiClient.get<AnnualReport[]>('/api/community-programs/annual-reports'),
  sponsorshipPackages: () => apiClient.get<SponsorshipPackage[]>('/api/community-programs/sponsorship-packages'),
  myBusinesses: () => apiClient.get<CommunityBusiness[]>('/api/community-programs/member/businesses'),
  createBusiness: (data: BusinessInput) => apiClient.post<CommunityBusiness>('/api/community-programs/member/businesses', data),
  updateBusiness: (id: string, data: BusinessInput) => apiClient.put<CommunityBusiness>(`/api/community-programs/member/businesses/${id}`, data),
  journey: () => apiClient.get<NewcomerJourney | null>('/api/community-programs/member/newcomer'),
  saveJourney: (data: JourneyInput) => apiClient.put<NewcomerJourney>('/api/community-programs/member/newcomer', data),
  family: () => apiClient.get<FamilyHousehold | null>('/api/community-programs/member/family'),
  saveFamily: (householdName: string) => apiClient.put<FamilyHousehold>('/api/community-programs/member/family', { householdName }),
  addFamilyMember: (data: { fullName: string; relationship: string; email?: string; birthDate?: string }) => apiClient.post<FamilyHousehold>('/api/community-programs/member/family/members', data),
  removeFamilyMember: (id: string) => apiClient.delete<FamilyHousehold>(`/api/community-programs/member/family/members/${id}`),
  appointmentSlots: () => apiClient.get<AppointmentSlot[]>('/api/community-programs/member/appointments/slots'),
  myAppointments: () => apiClient.get<AppointmentBooking[]>('/api/community-programs/member/appointments'),
  bookAppointment: (slotId: string, reason?: string) => apiClient.post<AppointmentBooking>('/api/community-programs/member/appointments', { slotId, reason }),
  cancelAppointment: (id: string) => apiClient.post<AppointmentBooking>(`/api/community-programs/member/appointments/${id}/cancel`),
  benefits: () => apiClient.get<PartnerBenefit[]>('/api/community-programs/member/benefits'),
  claimBenefit: (id: string) => apiClient.post<PartnerBenefit>(`/api/community-programs/member/benefits/${id}/claim`),
  myGrantApplications: () => apiClient.get<GrantApplication[]>('/api/community-programs/member/grant-applications'),
  applyForGrant: (data: { grantProgramId: string; applicantName: string; applicantEmail: string; statement: string; answers?: Record<string, string>; documents?: string[] }) => apiClient.post<GrantApplication>('/api/community-programs/member/grant-applications', data),
  withdrawGrant: (id: string) => apiClient.post<GrantApplication>(`/api/community-programs/member/grant-applications/${id}/withdraw`),
  mySponsorships: () => apiClient.get<SponsorshipRequest[]>('/api/community-programs/member/sponsorships'),
  requestSponsorship: (data: { packageId?: string; organizationName: string; contactEmail: string; objective: string; proposedAmountCents: number; currency: string }) => apiClient.post<SponsorshipRequest>('/api/community-programs/member/sponsorships', data),
  adminOverview: () => apiClient.get<ProgramsAdminOverview>('/api/admin/community-programs/overview'),
  reviewBusiness: (id: string, status: string, isFeatured = false, reviewNotes?: string) => apiClient.patch<CommunityBusiness>(`/api/admin/community-programs/businesses/${id}`, { status, isFeatured, reviewNotes }),
  saveOffering: (data: Omit<AppointmentOffering, 'id'>, id?: string) => id ? apiClient.put<AppointmentOffering>(`/api/admin/community-programs/appointment-offerings/${id}`, data) : apiClient.post<AppointmentOffering>('/api/admin/community-programs/appointment-offerings', data),
  createSlot: (data: { offeringId: string; startsAtUtc: string; endsAtUtc: string; capacity: number }) => apiClient.post<AppointmentSlot>('/api/admin/community-programs/appointment-slots', data),
  saveBenefit: (data: Record<string, unknown>, id?: string) => id ? apiClient.put<PartnerBenefit>(`/api/admin/community-programs/benefits/${id}`, data) : apiClient.post<PartnerBenefit>('/api/admin/community-programs/benefits', data),
  reviewGrant: (id: string, status: string, adminNotes?: string) => apiClient.patch<GrantApplication>(`/api/admin/community-programs/grant-applications/${id}`, { status, adminNotes }),
  saveSponsorshipPackage: (data: Record<string, unknown>, id?: string) => id ? apiClient.put<SponsorshipPackage>(`/api/admin/community-programs/sponsorship-packages/${id}`, data) : apiClient.post<SponsorshipPackage>('/api/admin/community-programs/sponsorship-packages', data),
  reviewSponsorship: (id: string, status: string, notes?: string) => apiClient.patch<SponsorshipRequest>(`/api/admin/community-programs/sponsorships/${id}`, { status, notes }),
  generateReport: (year: number) => apiClient.post<AnnualReport>(`/api/admin/community-programs/annual-reports/${year}/generate`),
  publishReport: (id: string) => apiClient.post<AnnualReport>(`/api/admin/community-programs/annual-reports/${id}/publish`),
  updateAutomation: (id: string, isEnabled: boolean, cadence: string) => apiClient.put<AutomationRule>(`/api/admin/community-programs/automations/${id}`, { isEnabled, cadence }),
  runAutomations: () => apiClient.post<AutomationRule[]>('/api/admin/community-programs/automations/run'),
};
